source/adios2/engine/sst/SstReader.cpp

```diff
@@ -58,8 +58,8 @@ SstReader::SstReader(IO &io, const std::string &name, const Mode mode,
         std::vector<size_t> VecStart;
         std::vector<size_t> VecCount;
         std::string Type(type);
-        SstReader::SstReader *Reader =
-            reinterpret_cast<SstReader::SstReader *>(reader);
+        typename SstReader::SstReader *Reader =
+            reinterpret_cast<typename SstReader::SstReader *>(reader);
```

Comment:
Why would this need `typename` when there are no template arguments.

Reply:
My Ubuntu 16.04, g++ 5.4.0 did not like that line. ```typename``` fixed it. Is there a better fix?